docs/arch_develop.md

```diff
@@ -71,39 +71,35 @@ Some components of the core library depend on the `package.json`s of the extensi
     -   Does not restore, it is a superset of what exists in `packages/core` for `configuration.properties`.
     -   To develop for the Amazon Q extension: add all changes to `packages/amazonq/package.json`, EXCEPT for settings that are references by code in the core library, or settings that already exist in the core `package.json`
 
-## Shared vs Common names
+## `web`, `node`, `common`, `shared` naming conventions
 
-In this repo, the keywords **"shared"** and **"common"** have specific meanings in the context of file/folder names.
+This project can run in different environments, eg Web mode (in the browser with no compute backend), or in Node.js on your desktop (the most common way).
+A problem arises when we use code that is exclusive to one environment, an example being Node.js' Filesystem module which will fail in Web mode.
 
-### "common"
+To ensure developers use compatible code for their environment we have subfolders in each topic which contains environment specific code in a single place.
 
-Code within a folder/file that has the "common" keyword implies that it can run in any environment. Examples of environments are: Web mode, or Node.js (local desktop)
+Using this file tree as reference, here are the rules:
 
-We need this distinction since not all code can run in any environment. A common example is filesystem code, where the actual implementation used could work in Node.js but not in Web mode.
-
-### "shared"
-
-Code within a folder/file that has the "shared" keyword implies that it is intended to be reused wherever it can be. This is generalized code that "Feature A" or "Feature B" could use if it works for their use case.
-
-An example is the `waitUntil()` function which continuously invokes an arbitrary callback function until it succeeds.
-
-> NOTE: Something that is "shared" does not mean it is "common", as it could be reused in different places but only work in Node.js for example.
+```
+src/
+├── myTopic/
+│   ├── commonCode.ts
+│   ├── node/
+│   │   └── nodeCode.ts
+│   └── web/
+│       └── webCode.ts
```

Comment:
for cases where a `node/` folder is needed, we wouldn't add "node" in the file names.

```suggestion
│   ├── subtopic.ts
│   ├── node/
│   │   └── subtopic.ts
│   └── web/
│       └── subtopic.ts
```

Reply:
My intention was that the actual file name is up to the user. Will update with your changes as it is more clear